src/systems/filecoin_blockchain/struct/block/_index.md

```diff
@@ -7,5 +7,66 @@ The Block is a unit of the Filecoin blockchain.
 
 {{< readfile file="block.id" code="true" lang="go" >}}
 
-{{< readfile file="block.go" code="true" lang="go" >}}
+{{< readfile file="election.id" code="true" lang="go" >}}
+
+# Block syntax validation
+
+Syntax validation refers to validation that may be performed on a block and its messages 
+without reference to outside information such as the parent state tree.
+
+An invalid block must not be transmitted or referenced as a parent.
+
+A syntactically valid block header must decode into fields matching the type definition below. 
+
+A syntactically valid header must have:
+
+- a non-empty array of `Parents` CIDs if `Epoch` is greater than zero,
+- a non-negative `ParentWeight`,
+- a `Miner` address which is an ID-address,
+- a non-negative `Epoch`,
+- a positive `Timestamp`,
+- a `Ticket` with non-empty `VRFResult`,
+- an `ElectionProof` with non-empty `VRFResult`.
+
+A syntactically valid full block must have:
+
+- all referenced messages syntactically valid,
+- all referenced parent receipts syntactically valid,
+
+Note that validation of the block signature requires access to the miner worker address from the
+parent tipset state, so signature validation forms part of sematic validation. Similarly, message 
+signature validation requires lookup of the public key associated with each messages `From`
+account actor in the block's parent state.
+
+# Block semantic validation
+
+Semantic validation refers to validation that requires reference to information outside the block
+header and messages themselves, in particular the parent tipset and state on which the block is built.
+
+A semantically valid block must have:
+
+- `Parents` listed in lexicographic order of their header's `Ticket`,
+- `Parents` all reference valid blocks and form a valid tipset,
+- `ParentStateRoot` identifying the state tree produced by executing the parent tipset's messages,
```

Comment:
add an extra line specifically mentioning that val here means repeating the message processing of grandparent messages (per vm interpreter) to yield `ParentStateRoot` and then comparing both state trees